sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1254,10 +1260,11 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
         ctx.addReferenceObj("zoneId", zoneId, zoneIdClass.getName),
         zoneIdClass)
       val longOpt = ctx.freshVariable("longOpt", classOf[Option[Long]])
+      val stringToTimestampFunc = if (ansiEnabled) "stringToTimestampAnsi" else "stringToTimestamp"
```

Comment:
If `stringToTimestampAnsi`  returns `Long` directly, the codegen for it will be largely simplified.

Reply:
done.